Sources/SpeziViews/Views/Controls/OptionSetPicker.swift

```diff
@@ -0,0 +1,229 @@
+//
+// This source file is part of the Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+
+/// The view style for an `OptionSetPicker`.
+public enum OptionSetPickerStyle {
+    /// The picker is rendered inline (e.g., in a List view).
+    case inline
+    /// The picker is rendered as a menu.
+    case menu
+}
+
+
+@available(iOS 18.0, macOS 15.0, tvOS 18.0, watchOS 11.0, visionOS 2.0, *)
+private struct ViewBasedOnVisibility<Unlabeled: View, Labeled: View>: View {
+    private let unlabeled: Unlabeled
+    private let labeled: Labeled
+
+    @Environment(\.labelsVisibility)
+    private var labelsVisibility
+
+
+    var body: some View {
+        switch labelsVisibility {
+        case .hidden:
+            unlabeled
+        case .automatic, .visible:
+            labeled
+        }
+    }
+
+    init(@ViewBuilder unlabeled: () -> Unlabeled, @ViewBuilder labeled: () -> Labeled) {
+        self.unlabeled = unlabeled()
+        self.labeled = labeled()
+    }
+}
+
+
+/// Create a picker based on a `OptionSet` selection.
+///
+/// If you have a type that both conforms to [`OptionSet`](https://developer.apple.com/documentation/swift/optionset)  and
+/// ``PickerValue`` you can use this Picker for your `selection` value.
+///
+/// `OptionSet` by definition allows the selection of multiple values.
+///
+/// - Note: Displaying labels is only supported on iOS 18 and newer.
+///
+/// ## Topics
+///
+/// ### Styling
+/// - ``OptionSetPickerStyle``
+public struct OptionSetPicker<Label: View, Value: OptionSet & PickerValue>: View
+    where Value.AllCases: RandomAccessCollection, Value == Value.Element {
+    private let allowEmptySelection: Bool
+    private let style: OptionSetPickerStyle
+
+    private let label: Label
+
+    @Binding private var selection: Value
+
+    private var selectionCount: Int {
+        Value.allCases.count { value in
+            selection.contains(value)
+        }
+    }
+
+    private var singleSelection: Value? {
+        Value.allCases.first { value in
+            selection.contains(value)
+        }
+    }
+
+    public var body: some View {
+        switch style {
+        case .inline:
+            let view = ForEach(Value.allCases, id: \.self) { value in
+                button(for: value)
+            }
+
+            if #available(iOS 18.0, macOS 15.0, tvOS 18.0, watchOS 11.0, visionOS 2.0, *) {
+                ViewBasedOnVisibility {
+                    view
+                } labeled: {
+                    Section {
+                        view
+                    } header: {
+                        label
+                    }
+                }
+            } else {
+                view
+            }
+        case .menu:
+            Menu {
```

Comment:
Menu seems to be unavailable on watchOS, we might need to add some #available elements to this & the menu option for watchOS.